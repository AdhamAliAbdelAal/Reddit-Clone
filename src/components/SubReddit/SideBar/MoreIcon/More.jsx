import { Box } from '@mui/material';
import MoreHorizIcon from '@mui/icons-material/MoreHoriz';
import { useState } from 'react';
import ClickAwayListener from '@mui/material/ClickAwayListener';
import {
  Select, SelectBox, SelectItem,
} from './style';

export default function Sort() {
  const [showList1, setShowList1] = useState(false);

  // handle disable the list when click away
  const handleClick1 = () => {
    setShowList1((prev) => !prev);
  };

  const handleClickAway1 = () => {
    setShowList1(false);
  };

  return (
    <>
      <Select>
        <ClickAwayListener onClickAway={handleClickAway1}>
          <Box sx={{ display: 'flex' }} onClick={() => { handleClick1(); }}>
            <MoreHorizIcon />
          </Box>
        </ClickAwayListener>
      </Select>
      {showList1 && (
      <SelectBox>
        <SelectItem>Add to Custom Feed</SelectItem>
        <SelectItem>Add to Favorites</SelectItem>
      </SelectBox>
      )}
    </>

  );
}
