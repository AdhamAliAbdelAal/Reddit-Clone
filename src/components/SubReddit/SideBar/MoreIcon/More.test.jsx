import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import More from './More';

// test snapshot
test('test snapshot', async () => {
  const tree = renderer.create(<More />).toJSON();
  expect(tree).toMatchSnapshot();
});
