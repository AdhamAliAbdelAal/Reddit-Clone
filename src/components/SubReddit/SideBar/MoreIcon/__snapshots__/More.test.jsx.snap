// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="MuiBox-root css-uwcr2l"
>
  <div
    className="MuiBox-root css-k008qs"
    data-testid="sort"
    onClick={[Function]}
    onTouchEnd={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="MoreHorizIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
      />
    </svg>
  </div>
</div>
`;
